src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,187 @@
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(targets={'python_library': PythonLibrary})
+
+  @classmethod
+  def rules(cls):
+    return (
+      *super().rules(),
+      *pylint_rules(),
+      *download_pex_bin.rules(),
+      *inject_init.rules(),
+      *pex.rules(),
+      *prepare_chrooted_python_sources.rules(),
+      *strip_source_root.rules(),
+      *python_native_code.rules(),
+      *subprocess_environment.rules(),
+      RootRule(PylintTarget),
+    )
+
+  def run_pylint(
+    self,
+    source_files: List[FileContent],
+    *,
+    config: Optional[str] = None,
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.pylint"]
+    if config:
+      # TODO: figure out how to get this file to exist...
+      self.create_file(relpath="pylintrc", contents=config)
+      args.append("--pylint-config=pylintrc")
+    if passthrough_args:
+      args.append(f"--pylint-args='{passthrough_args}'")
+    if skip:
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    target = PylintTarget(
+      PythonTargetAdaptor(
+        sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+        address=Address.parse(f"{self.source_root}:target"),
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
```

Comment:
You might want to extract this out into a method called `create_python_library`. It looks like every test uses the same `path` and `target_type`. Usually, it's the same `name` but sometimes changes, so you could use a default value.